src/main/java/org/folio/circulation/resources/CheckInByBarcodeResource.java

```diff
@@ -100,9 +54,10 @@ private void checkin(RoutingContext routingContext) {
         processAdapter::getRequestQueue, CheckInProcessRecords::withRequestQueue))
       .thenApply(findRequestQueueResult -> findRequestQueueResult.map(
         processAdapter::setInHouseUse))
-      .thenComposeAsync(inHouseUseResult -> inHouseUseResult.combineAfter(
+      .thenApplyAsync(r -> r.map(records -> records.withLoggedInUserId(context.getUserId())))
+      .thenCompose(setUserResult -> setUserResult.after(processAdapter::logCheckInOperation))
```

Comment:
How was it determined that this would happen after the loan and the request queue are fetched?

Reply:
If the API is going to be used for logging in-house then this is the only place where we have enough data.

Reply:
What information is needed for in-house use?

Could this be addressed by moving the pre-requisites for this to earlier it the process flow?

Given that it is not clear if in-house use will be recorded during the check-in, I think the system should try to do this at the earliest point it can for the moment.

Reply:
For in-house use we need item with location, check-in service point and request queue.

Reply:
Ok, could those be the first kinds of records that are fetched in the check in process and then the check in record stored immediately after those.

Reply:
Sure, moved upper.